pkg/asset/agent/image/ignition.go

```diff
@@ -0,0 +1,47 @@
+package image
+
+import (
+	"encoding/json"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/openshift/installer/pkg/agent/imagebuilder"
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// Ignition is an asset that generates the agent installer ignition file.
+type Ignition struct {
+	Config *igntypes.Config
+}
+
+// Name returns the human-friendly name of the asset.
+func (a *Ignition) Name() string {
+	return "Agent Installer Ignition"
+}
+
+// Dependencies returns the assets on which the Ignition asset depends.
+func (a *Ignition) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the agent installer ignition.
+func (a *Ignition) Generate(dependencies asset.Parents) error {
+	configBuilder, err := imagebuilder.New()
+	if err != nil {
+		return err
+	}
+
+	ignitionBytes, err := configBuilder.Ignition()
+	if err != nil {
+		return err
+	}
+
+	ignition := &igntypes.Config{}
+
+	err = json.Unmarshal(ignitionBytes, ignition)
```

Comment:
Why unmarshal into a struct only to marshal back to text again in the next Asset?

Reply:
I was split between doing that or breaking out content.Ignition() to return just the config. I'll convert it to the later.

Reply:
I meant that Ignition.Config could be a `[]bytes` and then no conversion would be required.
This won't hurt anything though.